core/src/plugins/filed/python/python-fd.cc

```diff
@@ -264,17 +379,21 @@ static bRC freePlugin(PluginContext* plugin_ctx)
   if (plugin_priv_ctx->object) { free(plugin_priv_ctx->object); }
 
   // Stop any sub interpreter started per plugin instance.
-  PyEval_AcquireThread(plugin_priv_ctx->interpreter);
-
+  auto* ts = PopThreadStateForInterp(plugin_priv_ctx->interp);
+  PyEval_AcquireThread(ts);
 
   if (plugin_priv_ctx->pModule) { Py_DECREF(plugin_priv_ctx->pModule); }
 
-  Py_EndInterpreter(plugin_priv_ctx->interpreter);
+  // releases the gil since 3.12
+  Py_EndInterpreter(ts);
 #if PY_VERSION_HEX < VERSION_HEX(3, 2, 0)
   PyEval_ReleaseLock();
 #else
-  PyThreadState_Swap(mainThreadState);
-  PyEval_ReleaseThread(mainThreadState);
+  if (PyVersion() < VERSION_HEX(3, 12, 0)) {
+    // endinterpreter releases the gil since 3.12
+    PyThreadState_Swap(mainThreadState);
+    PyEval_ReleaseThread(mainThreadState);
+  }
 #endif
```

Comment:
I'd rather not mix `#if` and `if` if it is avoidable.
Maybe we can just do the following to avoid the preprocessor.
```
if (PyVersion() < VERSION_HEX(3, 2, 0) {
  PyEval_ReleaseLock();
} else if (PyVersion() < VERSION_HEX(3, 12, 0)) {
  // endinterpreter releases the gil since 3.12
  PyThreadState_Swap(mainThreadState);
  PyEval_ReleaseThread(mainThreadState);
}
```

Reply:
done

Reply:
This sadly does not work. Bareos will not compile on -O0 like this, since it will not eliminate the dead if clause and then complain that we use a deprecated function.  I instead removed the python3.2 part since we (hopefully!) do not support 3.2 anymore